packages/astro/src/prefetch/index.ts

```diff
@@ -215,28 +219,27 @@ export function prefetch(url: string, opts?: PrefetchOptions) {
 	if (!canPrefetchUrl(url, ignoreSlowConnection)) return;
 	prefetchedUrls.add(url);
 
-	const priority = opts?.with ?? 'link';
-	debug?.(`[astro] Prefetching ${url} with ${priority}`);
-
-	if (
-		clientPrerender &&
-		HTMLScriptElement.supports &&
-		HTMLScriptElement.supports('speculationrules')
-	) {
-		// this code is tree-shaken if unused
+	// Prefetch with speculationrules if `clientPrerender` is enabled and supported
+	// NOTE: This condition is tree-shaken if `clientPrerender` is false as its a static value
+	if (clientPrerender && HTMLScriptElement.supports?.('speculationrules')) {
+		debug?.(`[astro] Prefetching ${url} with <script type="speculationrules">`);
 		appendSpeculationRules(url);
-	} else if (priority === 'link') {
+	}
+	// Prefetch with link if supported
+	else if (
+		document.createElement('link').relList?.supports?.('prefetch') &&
```

Comment:
Isn't it somewhat expensive to do this on every link? Could it be done once?

Reply:
I benchmarked this locally and there isn't a big difference in perf from what I can tell compared to caching, so I kept this direct approach, and to save some bytes.